# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/util/ActiveElementManager.cpp
# Commit: 7f6e3a882896
# Full Hash: 7f6e3a88289637190503adc7f2acec348709ecc1
# Author: Dan Robertson <dan@dlrobertson.com>
# Date: 2023-09-27 21:24:33
# Description:
#   Bug 1855289 - Do not start delayed clear active state timer if it is null. r=botond
#   
#   If the delayed clear active state timer is null, do not attempt to start
#   the timer. In the case that the delayed clear active state timer is
#   null, the active state should be cleared on ProcessSingleTap().
# ==============================================================================

diff -r dcf55687fbb1 -r 7f6e3a882896 gfx/layers/apz/util/ActiveElementManager.cpp
--- a/gfx/layers/apz/util/ActiveElementManager.cpp	Wed Sep 27 15:24:44 2023 +0000
+++ b/gfx/layers/apz/util/ActiveElementManager.cpp	Wed Sep 27 15:34:52 2023 +0000
@@ -107,6 +107,10 @@
 
 void DelayedClearElementActivation::StartTimer() {
   MOZ_ASSERT(mTimer);
+  // If the timer is null, active content state has already been cleared.
+  if (!mTimer) {
+    return;
+  }
   nsresult rv = mTimer->InitWithCallback(
       this, StaticPrefs::ui_touch_activation_duration_ms(),
       nsITimer::TYPE_ONE_SHOT);
